python/ray/log_monitor.py

```diff
@@ -170,15 +170,30 @@ def check_log_files_and_publish_updates(self):
             assert not file_info.file_handle.closed
 
             lines_to_publish = []
+            driver_switches = []
+            if file_info.driver_id != None:
+                driver_switches.append((-1, file_info.driver_id))
+
             max_num_lines_to_read = 100
-            for _ in range(max_num_lines_to_read):
+            for idx in range(max_num_lines_to_read):
                 next_line = file_info.file_handle.readline()
                 if next_line == "":
                     break
                 if next_line[-1] == "\n":
                     next_line = next_line[:-1]
+
+                # Record when a worker switches drivers
+                if next_line.startswith("Ray driver id: "):
+                    driver_id = next_line.split(" ")[-1]
```

Comment:
One way that things could go wrong (no need to address this right now, I'm not actually sure how we would address that other than having separate workers for different drivers), is if a task starts a background thread that just runs forever and prints stuff in the background. Then that print statement could corrupt the `"Ray driver id: " print statement and then mess up what the log monitor does.